### DISCUSSION TITLE
Replace macro names of board variants containing dashes (-) with underscores (_)

### DISCUSSION THREAD

**Discussion by supcik** [espressif/arduino-esp32#7466] [Ideas]
There are still many Arduino [board variants](https://github.com/espressif/arduino-esp32/tree/master/variants/) with a dash in the name. For example, "ttgo-lora32-v1" or "lolin32-lite".

In [CMakeLists.txt](https://github.com/espressif/arduino-esp32/blob/9762b2392ad3928d8643911294e505e78fcc5b2f/CMakeLists.txt#L219), this variant name is capitalized and then a macro `-DARDUINO_${idf_target_caps}_DEV` is defined. However, the C preprocessor does not like macro names with dashes. According to the [documentation](https://gcc.gnu.org/onlinedocs/gcc-12.2.0/cpp/Tokenization.html#Tokenization) : _"An identifier is the same as an identifier in C: any sequence of letters, digits, or underscores, which begins with a letter or underscore."_

So, my suggestion is to replace the dashed (-) with an underscore (_) in the macro definition. This could be implemented in the CMakeLists.txt file as :

```cmake
string(REPLACE "-" "_" idf_target_for_macro "${idf_target_caps}")
target_compile_options(${COMPONENT_TARGET} PUBLIC
    ...
    -DARDUINO_${idf_target_for_macro}_DEV
    ...
```

I already made some tests, and it seems to work. Does someone see a problem with this idea? If no, I would be happy to propose a pull request with this change.

Note that this problem was also addresses in PlatformIO : https://github.com/platformio/platform-espressif32/issues/81

**Comment by VojtechBartoska** [comment]
Any feedback to this @SuGlider / @me-no-dev?

**Comment by supcik** [comment]
Shall I create an issue from this discussion and offer a Pull Request?

**Reply by me-no-dev** [reply]
yes, please do so :)

**Reply by supcik** [reply]
OK. See issue #7499.

**Reply by supcik** [reply]
and PR #7500
